### PR TITLE
#1625 fix transformer not picked up

### PR DESCRIPTION
**Pull request by borissmidt**
fixes related issue #1625 
It was easy to resolve the missing package context file by passing all files that need to be generated to protoc at once.
This is similar how sbt and gradle use protoc to compile the sources.

### CODE REVIEW

File: contrib/scalapblib/src/ScalaPBWorker.scala [hunk 1]
```diff
@@ -25,17 +25,18 @@ class ScalaPBWorker {
...
+            ): Unit = {
             val opts = if (scalaPBOptions.isEmpty) "" else scalaPBOptions + ":"
             val args = otherArgs ++ Seq(
               s"--scala_out=${opts}${generatedDirectory.getCanonicalPath}",
-              s"--proto_path=${source.getParentFile.getCanonicalPath}",
-              source.getCanonicalPath
-            )
+              s"--proto_path=${root.getCanonicalPath}"
+            ) ++ sources.map(_.getCanonicalPath)
+            println(args)
```
Comment (lefou) [inline contrib/scalapblib/src/ScalaPBWorker.scala:39]: Looks like a debugging left over. If not, please use an appropriate method or stream of `ctx.log`, e.g. `debug` for debug info.

**Review by lefou — COMMENTED** [review]
Thanks for your contribution! Please avoid reformatting unchanged lines, but if, please use Scalafmt with our config.

File: contrib/scalapblib/src/ScalaPBWorker.scala (hunk 1 above)
Comment (borissmidt) [reply contrib/scalapblib/src/ScalaPBWorker.scala]: Good catch, I've made it a debug log i think it is useful to see the exact command parameters passed to the generator then you can run always run it manually to find out what is 'wrong'.

File: contrib/scalapblib/src/ScalaPBWorker.scala [hunk 2]
```diff
@@ -25,17 +25,18 @@ class ScalaPBWorker {
...
           ): Unit = {
             val opts = if (scalaPBOptions.isEmpty) "" else scalaPBOptions + ":"
             val args = otherArgs ++ Seq(
               s"--scala_out=${opts}${generatedDirectory.getCanonicalPath}",
-              s"--proto_path=${source.getParentFile.getCanonicalPath}",
-              source.getCanonicalPath
-            )
+              s"--proto_path=${root.getCanonicalPath}"
+            ) ++ sources.map(_.getCanonicalPath)
+            ctx.log.debug(args.mkString(" "))
```
Comment (lefou) [inline contrib/scalapblib/src/ScalaPBWorker.scala:39]: I would prefer to have a more meaningful message, e.g. 

```suggestion
            ctx.log.debug("ScalaPBC args: ${args.mkString(" ")}")
```

Comment (borissmidt) [reply contrib/scalapblib/src/ScalaPBWorker.scala]: Good idea thank you for the suggestion  i have added it.

**Review by lefou — APPROVED** [review]
Thanks. Looks good now! :+1:

**Comment by borissmidt** [comment]
I might come back with another PR because i have projects with multiple modules that depend on each and they all do scalaPB. 
So i'm not sure if this will handle 'transititive' scalapb options.

**Comment by borissmidt** [comment]
Ah damn i just found 1 bug i have to add 
```if(!files.isEmpty) {
          compiler.compileScalaPB(inputDir.toIO, files, scalaPBOptions, dest.toIO, scalaPBCExtraArgs)
        }
        ```
        otherwise we get an error if there is no file

**Comment by lefou** [comment]
@borissmidt Best is to add a test case to reproduce the bug and open a new PR with the test and it's fix.

**Comment by borissmidt** [comment]
Yes indeed i just couldn't fix my initial problem that scalapb.options on package scope level of other projects and probably jars are not picked up.
So i think this is a bigger problem within scalapb.

**Pull request merged**